# Goal of Project: predict user's depression score and identify each user’s life patterns

In [11]:
# List of required packages
import pandas as pd
import numpy as np
import seaborn as sns
import glob
import os 
import matplotlib.pyplot as plt

In [12]:
# Load data
device_df = pd.read_csv('device_uplink.csv')
user_information_df = pd.read_csv('user_information.csv')

# Data Exploration:

# 1: Data Sample 

In [13]:
# Sample data
data = device_df.sample(10)
print(data)

        Unnamed: 0  uplink_id  owner_id          client_time  tag_id  step  \
296502      296502    3434968       536  2021-02-16 19:20:38    10.0  8025   
153361      153361    1783776       574  2020-10-28 22:40:04     0.0  3750   
62286        62286    1080848       569  2020-09-15 14:05:25    19.0  2587   
90417        90417    1325977       232  2020-10-01 08:02:18     7.0   493   
310125      310125    3591041       230  2021-02-27 09:25:57    10.0  3655   
22133        22133     624816       544  2020-08-13 17:03:33    12.0  5471   
66566        66566    1121196       524  2020-09-18 18:47:20     2.0  4575   
167263      167263    1904392       547  2020-11-05 08:00:05     0.0    68   
51028        51028     968312       506  2020-09-06 16:50:08     0.0  6580   
140740      140740    1691598       552  2020-10-22 16:50:14     0.0  7280   

        battery_low  is_charge  tag_battery_low  
296502           18          0                0  
153361           68          0           

In [14]:
# Sample data
user = user_information_df.sample(10)
print(user)

    user_id  birth year  age sex                                  etc  \
12      523        1942   80   F                       Not applicable   
27      582        1948   74   F                      차상위, 혈압, 당뇨, 관절   
7       572        1940   82   F                       Not applicable   
6       496        1939   83   F            당뇨, 허리협착증, 저혈당, 갑자기 쓰러지기도   
3       495        1937   85   F                       Not applicable   
23      569        1947   75   F                       Not applicable   
30      661        1949   73   F                                 고지혈증   
25      508        1948   74   F                       Not applicable   
15      586        1944   78   F  이석증, 거리에서 두번 정신잃고 쓰러져 구급차에 실려 간적 있음   
50      668        1948   74   M                               당뇨, 혈압   

    depression_score   depression_class  
12             0.500           Moderate  
27             0.375               Mild  
7              0.500           Moderate  
6              0.750  Modera

# 2: Display dimension and Columns

In [15]:
#Assess dimension and display columns of device file
deviceshape = device_df.shape 
devicecol = device_df.columns 

print("the shape of the df is",deviceshape)
print("the columns are", devicecol)

the shape of the df is (324823, 9)
the columns are Index(['Unnamed: 0', 'uplink_id', 'owner_id', 'client_time', 'tag_id', 'step',
       'battery_low', 'is_charge', 'tag_battery_low'],
      dtype='object')


In [16]:
#Assess dimension and display columns of user information file
usershape = user_information_df.shape 
usercol = user_information_df.columns 

print("the shape of the df is",usershape)
print("the columns are", usercol)

the shape of the df is (53, 7)
the columns are Index(['user_id', 'birth year', 'age', 'sex', 'etc', 'depression_score',
       'depression_class'],
      dtype='object')


# 3: Check and Convert Data Types

In [17]:
device_df.dtypes

Unnamed: 0           int64
uplink_id            int64
owner_id             int64
client_time         object
tag_id             float64
step                 int64
battery_low          int64
is_charge            int64
tag_battery_low      int64
dtype: object

In [19]:
user_information_df.dtypes

user_id               int64
birth year            int64
age                   int64
sex                  object
etc                  object
depression_score    float64
depression_class     object
dtype: object

In [20]:
device_df['client_time'] = pd.to_datetime(device_df['client_time'])

In [21]:
device_df.dtypes

Unnamed: 0                  int64
uplink_id                   int64
owner_id                    int64
client_time        datetime64[ns]
tag_id                    float64
step                        int64
battery_low                 int64
is_charge                   int64
tag_battery_low             int64
dtype: object

# 4: Find and Replace Missing Values

In [22]:
device_df.isnull().sum()

Unnamed: 0            0
uplink_id             0
owner_id              0
client_time           0
tag_id             1206
step                  0
battery_low           0
is_charge             0
tag_battery_low       0
dtype: int64

In [23]:
user_information_df.isnull().sum()

user_id             0
birth year          0
age                 0
sex                 0
etc                 0
depression_score    0
depression_class    0
dtype: int64

# 5: Reduce Dimension

# 6: Visualize Data

# 7: Notes on Data

# 8: Fit a multiple linear regression model to the depression score

# 9: Using the estimated regression model, what depression score is predicted for user 495 and 496? What is the prediction error?

# 10: Use stepwise regression with the three options (backward, forward, both) to reduce the remaining predictors as follows: Run stepwise on the training set.

# 11: Run a naïve Bayes classifier on the training set with the relevant predictors

# 12: Compute the overall error, specificity, sensitivity, false-positive error, and false-negative error for the validation set.

# 13: Examine the conditional probabilities of the output. What is the probability of a user having ‘moderately severe’ depression when their average total steps per day is below 10000? 𝑃(𝐷𝐶 = "𝑀𝑜𝑑𝑒𝑟𝑎𝑡𝑒𝑙𝑦 𝑠𝑒𝑣𝑒𝑟𝑒”| (𝑎𝑣𝑒𝑟𝑎𝑔𝑒 𝑡𝑜𝑡𝑎𝑙 𝑠𝑡𝑒𝑝𝑠 𝑝𝑒𝑟 𝑑𝑎𝑦) ≤ 10000)?

# 14: Classification modeling with k-nearest neighbors classifier for daily activities.